tools/lib/provision.py

```diff
@@ -26,7 +26,7 @@
 
 if False:
     # `NoReturn` was introduced in python 3.5.4
-    from typing import List, NoReturn
+    from typing import List, NoReturn  # type: ignore # ->warn-unreachable
```

Comment:
Use [`if TYPE_CHECKING`](https://mypy.readthedocs.io/en/latest/common_issues.html#import-cycles).

Reply:
For provision and webpack, because we are conditionally importing from `typing`, it seems using a constant named `MYPY`, that's treated the same as `TYPE_CHECKING`, would be cleaner?

So we have something like:
```
MYPY = False
if MYPY:
    # `NoReturn` was introduced in python 3.5.4
    from typing import List, NoReturn
```

Instead of repeated imports like:

```
from typing import TYPE_CHECKING, list
if TYPE_CHECKING:
    from typing import NoReturn
```

Reply:
This should be converted to just 
`from mypy_extensions import NoReturn`; there's a standard solution for this particular issue.

Reply:
Ahh, I really need to learn to `grep` before I speak.  However, this didn't work.  If we import unconditionally, I still see:

```
Jul 31 02:03:00 Traceback (most recent call last):
Jul 31 02:03:00   File "./lib/provision.py", line 29, in <module>
Jul 31 02:03:00     from mypy_extensions import NoReturn
Jul 31 02:03:00 ModuleNotFoundError: No module named 'mypy_extensions'
Jul 31 02:03:00 
```

So, I updated to import like:
```
from typing import List, TYPE_CHECKING
if TYPE_CHECKING:
    # `NoReturn` wasn't introduced until python 3.5.4
    from mypy_extensions import NoReturn
```

Reply:
For `provision` in particular, we still need `if False` because our deps might not be installed yet; I think the others are fine.